### PR TITLE
IGNITE-22936 Sql. Improve error message when trying to use recursive CTE

### PR DESCRIPTION
**Pull request by korlov42**
https://issues.apache.org/jira/browse/IGNITE-22936

---
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/integrationTest/sql/basic_queries/not_supported_queries.test [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# name: test/sql/basic_queries/not_supported_queries.test
+# group: [basic_queries]
+
+statement error: Failed to validate query. From line 1, column 6 to line 1, column 14: A recursive query is not supported.
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/sql/basic_queries/not_supported_queries.test:4]: Let's add mention SQL Standart features T131(Recursive query) and T132 (Recursive query in subquery)
Also, we need to add statement for the T132 feature.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteResource.java [hunk 2]
```diff
@@ -39,62 +39,65 @@
...
-    @Resources.BaseMessage("Column N#{0} matched using NATURAL keyword or USING clause "
+    @BaseMessage("Column N#{0} matched using NATURAL keyword or USING clause "
             + "has incompatible types in this context: ''{1}'' to ''{2}''")
     Resources.ExInst<SqlValidatorException> naturalOrUsingColumnNotCompatible(int num, String type1, String type2);
 
     @BaseMessage("Cannot apply ''{0}'' to arguments of type {1}.")
     ExInst<SqlValidatorException> canNotApplyOp2Type(String a0, String a1);
 
+    @BaseMessage("A recursive query is not supported.")
+    ExInst<SqlValidatorException> recursiveQueryIsNotSupported();
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteResource.java:99]: Let's use single approach for whole class.
Ether we should use pattern `Resources.ExInst ...` or ``ExInst ...``

File: modules/sql-engine/src/integrationTest/sql/basic_queries/not_supported_queries.test (hunk 1 above)
Comment (korlov42) [reply modules/sql-engine/src/integrationTest/sql/basic_queries/not_supported_queries.test:5]: done

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteResource.java (hunk 2 above)
Comment (korlov42) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteResource.java:99]: chose the shortest option, done

**Review by ygerzhedovich — APPROVED** [review]

**Review by xtern — APPROVED** [review]

**Pull request merged**